download-episodes.py

```diff
@@ -0,0 +1,237 @@
+    verbose('Getting list of all downloaded episodes...')
+    download_folder = os.path.expanduser(_get_episode_downloads_folder())
+    os.chdir(download_folder)
+
+    files = glob('*.' + _get_file_format())
+
+    verbose(f'Found {len(files)} downloads in the folder...')
+    downloads = []
+    for file in files:
+        downloaded_file = Download(file)
+        downloads.append(downloaded_file)
+
+    return downloads
+
+
+def _parse_episodes(show):
+    parsed_episodes = []
+    for episode in show['episodes']:
+        parsed = Episode(episode)
+        parsed_episodes.append(parsed)
+
+    return parsed_episodes
+
+
+def get_latest_episodes():
+    show = _get_show_response()
+    verbose(f'Show json response: {json.dumps(show, indent=2)}')
+    episodes = _parse_episodes(show)
+    verbose(f'Found {len(episodes)} episodes in show response...')
+    for e in episodes:
+        e.print()
+    return episodes
+
+
+def download(episode):
+    if args.with_fake_response:
+        progress_complete(episode.title_with_date, 100)
+        return
+
+    url_request = request.Request(episode.stream_url)
+    url_connect = request.urlopen(url_request)
+
+    file_size = int(url_connect.info()['Content-Length'])
+    download_block_size = 8192
+    downloaded_size = 0
+
+    out_file_name = datetime.strftime(episode.date, "%Y%m%d") + '.' + _get_file_format()
+    download_folder = os.path.expanduser(_get_episode_downloads_folder())
+    tmp_folder = download_folder + 'tmp'
+
+    if not os.path.exists(tmp_folder):
+        os.mkdir(tmp_folder)
+    os.chdir(tmp_folder)
+
+    with open(out_file_name, 'wb') as out_file:
+        while True:
+            buffer = url_connect.read(download_block_size)
+            downloaded_size += len(buffer)
+            if not buffer:
+                break
+            out_file.write(buffer)
+            progress(episode.title_with_date, file_size, downloaded_size)
+
+    os.rename(out_file_name, download_folder + out_file_name)
+
+    progress_complete(episode.title_with_date, file_size)
+
+
+def progress(file_name, total_len, current_size):
+    max_blocks = 50
+    progress_blocks = int(total_len / max_blocks)
+    current_position = int(current_size / progress_blocks)
+    sys.stdout.write('\r%s: [%s%s] %s/%s %s'
+                     % (file_name,
+                        '=' * current_position,
+                        ' ' * (max_blocks - current_position),
+                        current_size,
+                        total_len,
+                        ''
+                        ))
+    sys.stdout.flush()
+
+
+def progress_complete(file_name, total_len):
+    max_blocks = 50
+    print('\r%s: [%s] %s/%s => ✅ Done' % (file_name, '=' * max_blocks, total_len, total_len))
+
+
+def get_episodes_to_download():
+    episodes = get_latest_episodes()
+    downloaded = get_downloaded_episodes()
+
+    episodes_to_download = []
+    for episode in episodes:
+        found = False
+        for file in downloaded:
```

Comment:
You should hoist this out of the loop and put the dates into a set.
e.g. `downloaded_dates = {file.date for file in downloaded}`

Then you can just do:
```
if episode.date not in downloaded_dates:
    episodes_to_download.append(episode)
```